e2e/e2etests/test_migrate_tss.go

```diff
@@ -44,32 +44,39 @@ func TestMigrateTSS(r *runner.E2ERunner, _ []string) {
 	}
 
 	btcTSSBalanceOld := btcBalance
+	halfBtcBalance := btcBalance / 2
 	// Use fixed fee of 0.02 for migration
 	// Code links for reference to calculate fee:
 	// https://github.com/zeta-chain/node/blob/5c73a45b6096ed7f7387ae8f7ccc7d5e2c668e0d/zetaclient/chains/bitcoin/signer/sign.go#L54
 	// https://github.com/zeta-chain/node/blob/5c73a45b6096ed7f7387ae8f7ccc7d5e2c668e0d/zetaclient/chains/bitcoin/signer/sign.go#L32
-	btcBalance = btcBalance - 0.02
+	halfBtcBalance = halfBtcBalance - 0.02
 	btcChain := chains.BitcoinRegtest.ChainId
 
 	r.WaitForTSSGeneration(2)
 
 	//migrate btc funds
 	// #nosec G701 e2eTest - always in range
-	migrationAmountBTC := sdkmath.NewUint(uint64(btcBalance * 1e8))
-	msgMigrateFunds := crosschaintypes.NewMsgMigrateTssFunds(
-		r.ZetaTxServer.MustGetAccountAddressFromName(utils.AdminPolicyName),
-		btcChain,
-		migrationAmountBTC,
-	)
-	_, err = r.ZetaTxServer.BroadcastTx(utils.AdminPolicyName, msgMigrateFunds)
-	require.NoError(r, err)
-
-	// Fetch migrator cctx for btc migration
-	migrator, err := r.ObserverClient.TssFundsMigratorInfo(r.Ctx, &observertypes.QueryTssFundsMigratorInfoRequest{
-		ChainId: btcChain})
-	require.NoError(r, err)
-	cctxBTCMigration := migrator.TssFundsMigrator.MigrationCctxIndex
-
+	cctxBTCAmount := sdkmath.ZeroUint()
+	for i := 0; i < 2; i++ {
```

Comment:
can we maybe add some comment explaining why we need to do this twice in a loop?